build-logic/jvm/src/main/kotlin/gradlebuild/jvm/extension/UnitTestAndCompileExtension.kt

```diff
@@ -51,7 +51,7 @@ abstract class UnitTestAndCompileExtension(
     private
     fun enforceCompatibility(majorVersion: Int) {
         tasks.withType<JavaCompile>().configureEach {
-            options.release.set(null as Int?)
+            options.release = null
```

Comment:
💬 Nice! How come we don't need the cast anymore?

Reply:
Since `.set` is written in Java, Kotlin can't resolve what method accepts nullable and what doesn't (even though we use `@Nullable` annotations. But since `.assign` extension methods are in Kotlin, we set T variant as nullable and `Provider<T>` variant as non-nullable and cast is not needed:
```
@Incubating
fun <T> Property<T>.assign(value: T?) {
    emitIncubatingLogMessage()
    this.set(value)
}

@Incubating
fun <T> Property<T>.assign(value: Provider<out T?>) {
    emitIncubatingLogMessage()
    this.set(value)
}
```

Note: as soon we will add new method, that accepts nullable, cast will be needed again.

Reply:
Cool!